draft-howard-rats-coserv.md

```diff
@@ -124,18 +156,70 @@ The transactional pattern between the producer and the consumer would be that th
+All three methods correspond to environments that are also defined within CoRIM {{-rats-corim}}.
+
+- **Class**: A class is an environment that is expected to be common to a group of similarly-constructed Attesters, who might therefore share the same set of endorsed characteristics.
+An example of this might be a fleet of computing devices of the same model and manufacturer.
+
+- **Instance**: An instance is an environment that is unique to an individual and identifiable Attester, such as a single computing device or component.
+
+- **Group**: A group is a collection of common Attester instances that are collected together based on some defined semantics.
+For example, Attesters may be put into groups for the purpose of anonymity.
 
-The purpose of a query is to allow the consumer (Verifier) to specify the artifacts (Endorsements and Reference Values) that it needs.
-Consequently, a query corresponds to the environmental characteristics of one or more Attesters.
-Such environmental characteristics are identical to those used in the information model of CoRIM {{-rats-corim}}.
-In summary, they can include the following:
+## Queries
 
-- An individual Attester instance.
-- A group (identifiable collection) of Attester instances.
-- A class of Attester, defined by characteristics such as the vendor or model, of which there may be an arbitrary number of instances.
+The purpose of a query is to allow the consumer (Verifier) to specify the artifacts that it needs.
+The information that is conveyed in a CoSERV query includes the following:
 
+- A specification of the required artifact type: Reference Value, Endorsed Value or Trust Anchor.
+See {{secartifacts}} for definitions of artifact types.
+A single CoSERV query can only specify a single artifact type.
+- A specification of the Attester's environment.
+Environments can be selected according to Attester instance, group or class.
+See {{secenvironments}} for full definitions.
 To facilitate efficient transactions, a single query can specify either multiple instances, multiple groups or multiple classes.
+However, it is not possible to mix instance-based selectors, group-based selectors and class-based selectors in a single query.
+- A timestamp, denoting the time at which the CoSERV query was sent.
+- A switch to select the desired supply chain depth.
+A CoSERV query can request collected artifacts, source artifacts, or both.
+This switch is especially relevant when the CoSERV query is fulfilled by an aggregator.
+The collected artifacts are intended for convenient consumption (according to the shallow trust model), while the source artifacts are principally useful for auditing (according to the deep trust model).
+It is possible for a query to select for source artifacts only, without the collected artifacts.
+This might happen when the consumer needs to inspect or audit artifacts from across the deep supply chain, while not requiring the convenience of the aggregated view.
```

Comment:
Another use case for “source-artifacts” is that of a pure broker, pulling from multiple upstream sources for its downstream aggregators.